### PR TITLE
Allow ItemAlteration Hardness to use non-integer alteration values

### LINKED ISSUE #13910: Feature Request: Allow ItemAlteration for hardness to have a non-integer value
This would be useful for multiply by 0.5 - see Gorum - Minor Curse.

### PR DESCRIPTION
**Pull request by DocSchlock**
Closes #13910 

Uses Math.trunc() to ensure Hardness is always an integer and abides by the round down rule.

### CODE REVIEW

**Comment by CarlosFdez** [comment]
I'm not the person to ask about this, but I'll take a look at this after v12 is out. Less fundamental things are also on freeze in the meantime.

File: src/module/rules/rule-element/item-alteration/schemas.ts [hunk 1]
```diff
@@ -292,7 +292,7 @@ const ITEM_ALTERATION_VALIDATORS = {
             required: true,
             choices: ["add", "downgrade", "multiply", "override", "remove", "subtract", "upgrade"],
         }),
-        value: new fields.NumberField({ required: true, integer: true, nullable: false, positive: true } as const),
+        value: new fields.NumberField({ required: true, integer: false, nullable: false, positive: true } as const),
```
Comment (stwlam) [inline src/module/rules/rule-element/item-alteration/schemas.ts:295]:
```suggestion
        value: new fields.NumberField({ required: true, nullable: false, min: 0 } as const),
```
We can let in 0 too

File: src/module/rules/rule-element/item-alteration/alteration.ts [hunk 2]
```diff
@@ -236,7 +236,7 @@ class ItemAlteration extends foundry.abstract.DataModel<RuleElementPF2e, ItemAlt
                     const system = data.item.system;
                     const value = data.alteration.value;
                     const newValue = AELikeRuleElement.getNewValue(this.mode, system.hardness, value);
-                    system.hardness = Math.max(newValue, 0);
+                    system.hardness = Math.max(Math.trunc(newValue), 0);
```
Comment (stwlam) [inline src/module/rules/rule-element/item-alteration/alteration.ts:239]:
```suggestion
                    system.hardness = Math.max(Math.floor(newValue), 0);
```

File: src/module/rules/rule-element/item-alteration/schemas.ts [hunk 3]
```diff
@@ -292,7 +292,7 @@ const ITEM_ALTERATION_VALIDATORS = {
             required: true,
             choices: ["add", "downgrade", "multiply", "override", "remove", "subtract", "upgrade"],
         }),
-        value: new fields.NumberField({ required: true, integer: true, nullable: false, positive: true } as const),
+        value: new fields.NumberField({ required: true, nullable: false, min: 0 } as const),
```
Comment (stwlam) [inline src/module/rules/rule-element/item-alteration/schemas.ts:295]:
```suggestion
        value: new fields.NumberField({ required: true, nullable: false, min: -20, max: 20 } as const),
```

File: src/module/rules/rule-element/item-alteration/alteration.ts [hunk 4]
```diff
@@ -236,7 +236,7 @@ class ItemAlteration extends foundry.abstract.DataModel<RuleElementPF2e, ItemAlt
                     const system = data.item.system;
                     const value = data.alteration.value;
                     const newValue = AELikeRuleElement.getNewValue(this.mode, system.hardness, value);
-                    system.hardness = Math.max(newValue, 0);
+                    system.hardness = Math.max(Math.floor(newValue), 0);
```
Comment (stwlam) [inline src/module/rules/rule-element/item-alteration/alteration.ts:239]:
```suggestion
                    system.hardness = Math.max(Math.trunc(newValue), 0);
```

**Pull request merged**